container.rs

```diff
@@ -80,45 +79,42 @@ pub trait Frame {
     fn rendering_offset(&self) -> i64;
 }
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub enum TrackType {
-    Video,
-    Audio,
-    Other,
+pub enum TrackType<'a> {
+    Video(Box<VideoTrack<'a> + 'a>),
+    Audio(Box<AudioTrack<'a> + 'a>),
+    Other(Box<Track<'a> + 'a>),
 }
 
+
 /// Generic convenience methods for tracks.
 pub trait TrackExt {
     fn debug(&self) -> String;
 }
-
-impl<'a,'b> TrackExt for &'a (Track + 'b) {
+/*
+impl<'a,'b> TrackExt for &'a (Track<'b> + 'b) {
```

Comment:
I don't really want to lose this functionality; it's useful for debugging. Can we at least move it into a method on `ContainerReader` that takes the track number?